manager/controllers/app/solve.go

```diff
@@ -45,6 +45,11 @@ func solveSingleDataset(env *datapath.Environment, dataset *datapath.DataInfo, l
 // find a solution for all data paths at once
 func solve(env *datapath.Environment, datasets []datapath.DataInfo, log *zerolog.Logger) ([]datapath.Solution, error) {
 	solutions := []datapath.Solution{}
+	if len(env.Modules) == 0 {
+		msg := "There are no deployed modules in the environment"
+		log.Error().Msg(msg)
+		return solutions, errors.New(msg)
+	}
```

Comment:
Since there are going to be more sanity checks, please move this check to a new function.

Reply:
Also, add the error message to a file that has already some error messages defined as constants, and add a unit test to verify that this error is propagated to the status. If there is already a unit test for this, just add a check that the error is as expected.